### DISCUSSION TITLE
Field "model_type" has conflict with protected namespace "model_".

### DISCUSSION THREAD

**Discussion by me-v2** [pydantic/pydantic#7121] [Pydantic V2]
```
from pydantic import BaseModel

class A(BaseModel):
    model_type: str = 'A'


print(A(model_type='B'))
```

UserWarning: Field "model_type" has conflict with protected namespace "model_".

You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
model_type='B'

**Comment by hramezani** [comment]
It is new in `Pydantic V2` and you can read more about it in [protected_namespace documentation](https://docs.pydantic.dev/latest/usage/model_config/#protected-namespaces).

**Comment by me-v2** [comment]
@haizaar  got this tricks. thanks.

**Comment by HongsCode** [comment]
same issue. 
Eventually I changed the variable to start with 'model_'.

**Reply by gcclinux** [reply]
in which file do you update that? I can't figure out where and how to fix that issue

**Reply by migueelst** [reply]
Had the same issue and solved it by following the hints given in the warning. You just need to add `model_config  = ConfigDict(protected_namespaces=())` on the classes that have fields starting with `model_`. For example:
``` python
from pydantic import BaseModel, ConfigDict

class A(BaseModel):
    model_config  = ConfigDict(protected_namespaces=())
    model_type: str = 'A'
```

Here's the link to the api docs to read more about [protected namespaces](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.protected_namespaces)

**Reply by VinhLoiIT** [reply]
This design is still vulnerable because people may already design their schema to have a field like "model_config." Now the field is owned by Pydantic v2. Also it cannot be distinguished between pydantic fields and actual model fields and we will need to look into the docs carefully or to wait until an error is raised

**Reply by dave-c-vt** [reply]
i agree with @VinhLoiIT. python already has a set of protected or restricted verbs and syntaxes. for pydantic to claim an entire subset of variable/attribute names as generic as "model_*" is ... well ... a reason to not use pydantic for codebases that deal with models.

**Comment by KIKNIM** [comment]
UserWarning: Field "   _the word  you need to find_    " has conflict with protected namespace "model_".
你只需要在全局里面找到the word  you need to find,找到问题变量名所在的文件的位置后在该文件中找到一个类似于
class BotUsage(BaseModel):
这样的类定义里面加上下面这个代码就行    
class Config:
        protected_namespaces = ()

**Comment by atlas-editor** [comment]
the downside of the `model_config['protected_namespaces'] = ()` solution is that if pydantic were to add a say `model_type` field internally to `BaseModel` then instead of raising an exception we only get a warning. does pydantic make any promises as to not add any new fields in certain releases? otherwise this solution may backfire
